### DISCUSSION TITLE
goodbye-spaservices-it-was-fun-while-it-lasted/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#57] [Blog comments]
# goodbye-spaservices-it-was-fun-while-it-lasted/

https://blogs.taiga.nl/martijn/posts/goodbye-spaservices-it-was-fun-while-it-lasted/

**Comment by disqus-comments-migrator** [comment]
<em>[Gary Kaganas](https://disqus.com/by/garykaganas/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 24 2020, 12:53</em>

---

<p>How do you reference the webpack built files? I'm using the <code>WriteFilePlugin</code> for the webpack dev-server, but having issues with HMR (I'm getting a <code>[HMR] Hot Module Replacement is disabled</code> message)</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 27 2020, 11:13, in reply to [Gary Kaganas](https://disqus.com/by/garykaganas/)'s [comment](https://github.com/martijnboland/blog/discussions/57#discussioncomment-8953059)</em>

---

<p>I just reference the main.js entry point and don't use the WriteFilePlugin. In development mode, the main.js file gets all the HMR functionality that sets up the web socket connection and serves the built files from memory. See also <a href="https://github.com/martijnboland/LeanAspNetCore-React" rel="nofollow noopener" title="https://github.com/martijnboland/LeanAspNetCore-React">https://github.com/martijnboland/LeanAspNetCore-React</a> for a working example.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Dalton Andrade](https://disqus.com/by/dalton_andrade/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 26 2020, 06:24</em>

---

<p>Mind blowing... It's a great Idea, but, how do that using vue-cli? Any ideas? thanks for this post!</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 27 2020, 10:55, in reply to [Dalton Andrade](https://disqus.com/by/dalton_andrade/)'s [comment](https://github.com/martijnboland/blog/discussions/57#discussioncomment-8953063)</em>

---

<p>I don't have any experience with vue-cli, but there is an option in vue.config.js to modify the webpack configuration directly: <a href="https://cli.vuejs.org/guide/webpack.html#simple-configuration" rel="nofollow noopener" title="https://cli.vuejs.org/guide/webpack.html#simple-configuration">https://cli.vuejs.org/guide/webpack.html#simple-configuration</a></p>

**Reply by disqus-comments-migrator** [reply]
<em>[Dalton Andrade](https://disqus.com/by/dalton_andrade/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 27 2020, 09:40, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/57#discussioncomment-8953064)</em>

---

<p>Yeah... I figured out a way to do this using the devServer proxy configuration straight into de vue.config.js and it worked like a charm!!</p><p>In my case, all requests to /federation proxies to my backend server, so I can use my asp-net backend and vue hot module reload toghether.</p><p><code><br>const isServing = process.argv[2].indexOf('serve') !== -1;<br>const path = require('path');<br>const fs = require('fs');<br><br>module.exports = {<br>    outputDir: '../wwwroot/client-app',<br>    filenameHashing: false,<br>    publicPath: isServing ? '/' : '/client-app',<br>    runtimeCompiler: true,<br>    configureWebpack: {<br>        devtool: 'source-map',<br>    },<br>    devServer: {<br>        host: 'sf8424.local.prodam',<br>        port: 443,<br>        https: {<br>            cert: fs.readFileSync(path.resolve(__dirname, 'sf8424.local.prodam.crt')),<br>            key: fs.readFileSync(path.resolve(__dirname, 'sf8424.local.prodam.key')),<br>            ca: fs.readFileSync(path.resolve(__dirname, 'sf8424.local.prodam-CA.pem')),<br>        },<br>        proxy: {<br>            '^/federation': {<br>                target: 'https://sf8424.local.prodam:44380',<br>                secure: false,<br>            }<br>        },<br>        disableHostCheck: true,<br>    },<br>    transpileDependencies: [<br>        'vuetify',<br>    ],<br>}<br></code></p>

**Comment by disqus-comments-migrator** [comment]
<em>[JeremyTCD](https://disqus.com/by/JeremyTCD/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 14 2020, 09:03</em>

---

<p>Hi Martijn, my org maintains a Microsoft.AspNetCore.NodeServices alternative, <a href="https://github.com/JeringTech/Javascript.NodeJS" rel="nofollow noopener" title="https://github.com/JeringTech/Javascript.NodeJS">Jering.Javascript.NodeJS</a>. It's got some extra features like concurrency and ability to invoke in-memory scripts. We've been asked to improve its discoverability, hopefully this comment helps anyone looking.</p><p>Smart alternative to webpack middleware you've got here by the way, thanks for writing this up!</p>

**Reply by disqus-comments-migrator** [reply]
<em>[quazecoatl](https://disqus.com/by/quazecoatl/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 10 2020, 08:28, in reply to [JeremyTCD](https://disqus.com/by/JeremyTCD/)'s [comment](https://github.com/martijnboland/blog/discussions/57#discussioncomment-8953067)</em>

---

<p>Hi Jeremy, Apparently Microsoft is still maintaining Microsoft.AspNetCore.NodeServices.See my other comment.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[quazecoatl](https://disqus.com/by/quazecoatl/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 10 2020, 08:16</em>

---

<p>Hi Martijn,</p><p>While the GitHub issue "announcing" the deprecation of  the <b>Microsoft.AspNetCore.NodeServices</b> is still open, if you look to the <b>NuGet </b>package releases of <b>Microsoft.AspNetCore.NodeServices</b>, we can see the package has been continuously maintained AND... there's a preview for <b>.NET 5</b>!</p><p>This is in contradiction with the statement in the Github issue: "These packages will later be removed in .NET 5".</p><p>Doesn't this mean they changed their mind? It looks very likely.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[JeremyTCD](https://disqus.com/by/JeremyTCD/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 11 2020, 03:20, in reply to [quazecoatl](https://disqus.com/by/quazecoatl/)'s [comment](https://github.com/martijnboland/blog/discussions/57#discussioncomment-8953070)</em>

---

<p>Hey quazecoatl, they haven't removed it (as of the .Net 5 previews), but they intend to eventually. As Dan Roth mentioned in the <a href="https://github.com/dotnet/aspnetcore/issues/12890#issue-477050062" rel="nofollow noopener" title="https://github.com/dotnet/aspnetcore/issues/12890#issue-477050062">Github issue</a>: "... These packages will <b>later be removed in .NET 5</b>".</p><p>Edit: The NodeServices Nuget package is auto-generated from the dotnet/aspnetcore repository, along with many other packages, whenever their shared version is bumped. Apart from project metadata, the package's underlying <a href="https://github.com/dotnet/aspnetcore/tree/master/src/Middleware/NodeServices/src" rel="nofollow noopener" title="https://github.com/dotnet/aspnetcore/tree/master/src/Middleware/NodeServices/src">source code has not been updated this last 10 months</a>. INodeServices <a href="https://github.com/dotnet/aspnetcore/blob/master/src/Middleware/NodeServices/src/INodeServices.cs" rel="nofollow noopener" title="https://github.com/dotnet/aspnetcore/blob/master/src/Middleware/NodeServices/src/INodeServices.cs"> has remained marked as obsolete</a>, with a "use spa extensions" warning.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[quazecoatl](https://disqus.com/by/quazecoatl/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 11 2020, 08:25, in reply to [JeremyTCD](https://disqus.com/by/JeremyTCD/)'s [comment](https://github.com/martijnboland/blog/discussions/57#discussioncomment-8953073)</em>

---

<p>Oh, wow! <br>I don't understand why they want to make it obsolete. Interoperability with Node JS is a big thing IMHO.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[JeremyTCD](https://disqus.com/by/JeremyTCD/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 11 2020, 10:18, in reply to [quazecoatl](https://disqus.com/by/quazecoatl/)'s [comment](https://github.com/martijnboland/blog/discussions/57#discussioncomment-8953074)</em>

---

<p>Yeah lol it's unfortunate, the Node.js ecosystem is massive. Oh well.</p>
